### DISCUSSION TITLE
help to build a apk

### DISCUSSION THREAD

**Discussion by angelayalac** [schwabe/ics-openvpn#1496] [Q&A]
Hello community, I want to compile the project again to build the apk with androidstudio but it won't let me compile, it throws me an error in the cmake, I wanted to see if someone would know what I can do to let me do the apk build
CMake Error at CMakeLists.txt:28 (include):
include could not find load file:
openssl/openssl.cmake

that comes from
CMake Error at CMakeLists.txt:28 (include):
include could not find load file:
openssl/openssl.cmake

CMake Warning (dev) in CMakeLists.txt:
    No project() command is present.  The top-level CMakeLists.txt file must
    contain a literal, direct call to the project() command.  Add a line of
    code such as
  
      project(ProjectName)
  
    near the top of the file, but after cmake_minimum_required().
  
    CMake is pretending there is a "project(Project)" command on the first
    line.
  This warning is for project developers.  Use -Wno-dev to suppress it.

**Comment by schwabe** [comment]
you need also to clone the submodules.

**Reply by angelayalac** [reply]
I´m new and I don´t know how to clone the submodules, can you help me passing me a video or link to learn :D
When I rebuild the project throws this
CMake Warning (dev) in CMakeLists.txt:
No project() command is present.  The top-level CMakeLists.txt file must
contain a literal, direct call to the project() command.  Add a line of
code such as
project(ProjectName)
near the top of the file, but after cmake_minimum_required().
CMake is pretending there is a "project(Project)" command on the first
line.
This warning is for project developers.  Use -Wno-dev to suppress it.

**Reply by schwabe** [reply]
Google should have ample tutorial and videos how to use git with submodules.

**Reply by angelayalac** [reply]
Thank you very much, sorry for so many questions to finish, would be all the submodules? , i would greatly appreciate your help.

**Reply by schwabe** [reply]
Yes, you need all submodules
